packages/machine/src/instructions.ts

```diff
@@ -41,109 +41,282 @@ export enum Opcode {
+          context,
+          node
+        );
+      },
+      Opcode.OP_SIGN,
+      (message: InternalMessage, context: Context) => {
+        const ret = NextMsgGenerator.generate2(
+          message.clientMessage,
+          context.intermediateResults.signature!
+        );
+        context.intermediateResults.outbox.push(ret);
+      },
+      Opcode.IO_SEND,
+      Opcode.IO_WAIT,
+      Opcode.OP_SIGN_VALIDATE,
+      Opcode.STATE_TRANSITION_COMMIT
+    ],
+    1: [
+      (message, context, node) => {
+        swap(message.clientMessage);
+        context.intermediateResults.proposedStateTransition = UninstallProposer.propose(
+          message,
+          node
+        );
+        context.intermediateResults.operation = EthOpGenerator.generate(
+          message,
+          () => {},
+          context,
+          node
+        );
+      },
+      Opcode.OP_SIGN_VALIDATE,
+      Opcode.OP_SIGN,
+      (message: InternalMessage, context: Context) => {
+        const ret = NextMsgGenerator.generate2(
+          message.clientMessage,
+          context.intermediateResults.signature!
+        );
+        context.intermediateResults.outbox.push(ret);
+      },
+      Opcode.IO_SEND,
+      Opcode.STATE_TRANSITION_COMMIT
+    ]
+  },
+  [cf.legacy.node.ActionName.INSTALL_METACHANNEL_APP]: {
+    0: [
+      (message: InternalMessage, context: Context, node) => {
+        // copy client message
+        context.intermediateResults.outbox.push(message.clientMessage);
+        context.intermediateResults.outbox[0].seq = 1;
+        context.intermediateResults.outbox[0].toAddress =
+          message.clientMessage.data.intermediary;
+      },
+
+      // send to intermediary
+      Opcode.IO_SEND
+    ],
+    1: [
+      (message: InternalMessage, context: Context, node) => {
+        const clientMessage = message.clientMessage;
+        context.intermediateResults.outbox.push(clientMessage);
+        context.intermediateResults.outbox[0].seq = 2;
+        context.intermediateResults.outbox[0].fromAddress =
+          clientMessage.data.initiating;
+        context.intermediateResults.outbox[0].toAddress =
+          clientMessage.data.responding;
+      },
+      Opcode.IO_SEND,
+      // wait for the install countersign
+      Opcode.IO_WAIT,
+      () => {}
+
+      // // send the self-remove
+      // Opcode.IO_SEND,
+      // Opcode.IO_SEND
+    ],
+    2: [
+      (message: InternalMessage, context: Context, node) => {
+        // countersign
+        const clientMessage = message.clientMessage;
+        context.intermediateResults.outbox.push(clientMessage);
+        context.intermediateResults.outbox[0].seq = 3;
+        context.intermediateResults.outbox[0].fromAddress =
+          clientMessage.data.responding;
+        context.intermediateResults.outbox[0].toAddress =
+          clientMessage.data.intermediary;
+      },
+      Opcode.IO_SEND
+
+      // // wait for self-remove
+      // Opcode.IO_WAIT
+    ]
+  }
```

Comment:
This file is kind of a monstrosity. Any ideas for how we could more clearly break this up? Perhaps into a file per "flow"?